Grey graph for no data/closed

Coreball opened the pull request:
### Summary <!-- Required -->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Itemize bug fixes, new features, and other changes -->
<!-- Feel free to break this into sub-sections, i.e. features, fixes, etc. -->
<!-- Some examples are shown below. -->

This pull request greys out bar hours for which there is no historical density data and standardizes graph behavior for when the facility is closed. With the change in what data is returned for historical requests on the back end, this should grey out graph hours that are not open on a particular day. The graph axis has also been tweaked to provide better contrast.

- [x] Added greyed out graph sections
- [x] Standardized graph behavior for when there is no data
- [x] Extracted out bar appearance configuration and removed redundant code
- [x] Fixed orange -> red limit from 0.75 to 0.85

### Test Plan <!-- Required -->

<!-- Provide screenshots or point out the additional unit tests -->

Appel is open when there are colored bars and closed when the hour positions are greyed out
<img width="573" alt="Screen Shot 2020-06-21 at 18 10 28" src="https://user-images.githubusercontent.com/22627336/85238562-007bf980-b3ec-11ea-9739-5abdd6c46ae2.png">

In this case, there is no historical data for Okenshields on Sundays and the dining hall is closed. Previously, "closed" would just appear centered in the middle of the graph. Change was made to be more in line with how regular days are displayed
<img width="573" alt="Screen Shot 2020-06-21 at 18 10 57" src="https://user-images.githubusercontent.com/22627336/85238556-f35f0a80-b3eb-11ea-8ded-7294ec1a67e7.png">

Coreball committed:
Grey bars to mask?

Coreball committed:
BarView distinguishing closed/open hours, fix 0.75 -> 0.85 limit

Coreball committed:
Standardize closed graph behavior and extract constraint setup

Coreball committed:
Separate out bar color configuration

Coreball committed:
Grey bar buffer above max bar height, remove unused const

Coreball committed:
Adjust axis coloring and height

AnshG714 commented on Campus Density/DiningCells/GraphCell.swift:
```diff
@@ -63,6 +62,11 @@ class GraphCell: UICollectionViewCell, UIGestureRecognizerDelegate {
     }
 
+    /// Returns the text description of density at a certain hour
+    static func descriptionLabelText(selectedHour: Int, densityMap: [Int: Double]) -> String {
```
Does this need to be a `static` function?

Coreball replied:
It's called in `PlacesDetailViewController+Extension` instead of repeating the same format string manually, but I'm not sure about best practices regarding where this function could be located -- I'll look into it a bit further

Coreball committed:
Simplify passing of graph description

Have the cell figure out the right string representation itself instead of doing that in the PlacesDetailVC for part of it

AnshG714 reviewed (approved):
Good job
